apollo-router/tests/metrics_tests.rs

```diff
@@ -43,8 +43,8 @@ async fn test_metrics_reloading() -> Result<(), BoxError> {
         router.assert_reloaded().await;
     }
 
-    router.assert_metrics_contains(r#"apollo_router_cache_hit_count_total{kind="query planner",service_name="apollo-router",storage="memory",otel_scope_name="apollo/router",otel_scope_version=""} 4"#, None).await;
-    router.assert_metrics_contains(r#"apollo_router_cache_miss_count_total{kind="query planner",service_name="apollo-router",storage="memory",otel_scope_name="apollo/router",otel_scope_version=""} 2"#, None).await;
+    router.assert_metrics_contains(r#"apollo_router_cache_hit_count_total{kind="query planner",storage="memory",otel_scope_name="apollo/router"} 4"#, None).await;
```

Comment:
should we add `service_name` in resources ?

Reply:
There's definitely an issue here. Investigating.

Reply:
OK, there's a couple of things that need fixing up on the code side, but it'll still not result in a change to the test because otel now puts service_name elsewhere:
```
target_info{service_name="apollo-router"} 1
```

This may be a breaking change for users. Let's discuss.

Reply:
I've improved this in: [0b8561f](https://github.com/apollographql/router/pull/3649/commits/0b8561f56f73e6d91cc158d06812fac2094ab320). But having service_name on metrics is a MUST NOT in the spec. 

https://opentelemetry.io/docs/specs/otel/compatibility/prometheus_and_openmetrics/#resource-attributes-1

Users may have to update their dashboards.